d2/src/main/java/com/linkedin/d2/jmx/RelativeLoadBalancerStrategyJmx.java

```diff
@@ -0,0 +1,168 @@
+import com.linkedin.d2.balancer.strategies.relative.TrackerClientState;
+import com.linkedin.d2.balancer.util.partitions.DefaultPartitionAccessor;
+import com.linkedin.util.degrader.CallTracker;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+
+public class RelativeLoadBalancerStrategyJmx implements RelativeLoadBalancerStrategyJmxMBean
+{
+  private final RelativeLoadBalancerStrategy _strategy;
+
+  public RelativeLoadBalancerStrategyJmx(RelativeLoadBalancerStrategy strategy)
+  {
+    _strategy = strategy;
+  }
+
+  @Override
+  public double getLatencyStandardDeviation()
+  {
+    Map<TrackerClient, TrackerClientState> stateMap =
+        _strategy.getPartitionState(DefaultPartitionAccessor.DEFAULT_PARTITION_ID).getTrackerClientStateMap();
+
+    List<Double> weightedLatencies = stateMap.keySet().stream()
+        .map(trackerClient -> getWeightedLatency(trackerClient.getLatestCallStats()))
+        .collect(Collectors.toList());
+
+    double avgLatency = weightedLatencies.stream()
+        .mapToDouble(Double::doubleValue)
+        .average()
+        .orElse(0);
+
+    double variance = weightedLatencies.stream()
+        .map(latency -> Math.pow(latency - avgLatency, 2))
+        .mapToDouble(Double::doubleValue)
+        .average()
+        .orElse(0);
+
+    return Math.sqrt(variance);
+  }
+
+  @Override
+  public double getLatencyAverageAbsoluteDeviation()
+  {
+    Map<TrackerClient, TrackerClientState> stateMap =
+        _strategy.getPartitionState(DefaultPartitionAccessor.DEFAULT_PARTITION_ID).getTrackerClientStateMap();
+
+    List<Double> weightedLatencies = stateMap.keySet().stream()
+        .map(trackerClient -> getWeightedLatency(trackerClient.getLatestCallStats()))
+        .collect(Collectors.toList());
+
+    double avgLatency = weightedLatencies.stream()
+        .mapToDouble(Double::doubleValue)
+        .average()
+        .orElse(0);
+
+    return weightedLatencies.stream()
+        .map(latency -> Math.abs(latency - avgLatency))
+        .mapToDouble(Double::doubleValue)
+        .average()
+        .orElse(0);
+  }
+
+  @Override
+  public double getLatencyMedianAbsoluteDeviation()
+  {
+    Map<TrackerClient, TrackerClientState> stateMap =
+        _strategy.getPartitionState(DefaultPartitionAccessor.DEFAULT_PARTITION_ID).getTrackerClientStateMap();
+
+    List<Double> weightedLatencies = stateMap.keySet().stream()
+        .map(trackerClient -> getWeightedLatency(trackerClient.getLatestCallStats()))
+        .collect(Collectors.toList());
+
+    double medianLatency = getMedian(weightedLatencies);
+
+    List<Double> medianAbsolutes = weightedLatencies.stream()
+        .map(latency -> Math.abs(latency - medianLatency))
+        .collect(Collectors.toList());
+
+    return getMedian(medianAbsolutes);
+  }
+
+  @Override
+  public double getLatencyMaxAbsoluteDeviation()
```

Comment:
On a second thought, I started thinking about for the max deviation, we probably care more about the ones that is greater than the average, if a host has 0 latency, it may impact this max absolute deviation. Should we not use absolute?

Reply:
Agreed. 

Maybe we could have
1. Standard Deviation and MeanAbsoluteDeviation to capture the entire variance
2. MeanSignedDeviation and MaxSignedDeviation to capture the positive outliers
3. We could also have a metric for (maxLatency - avgLatency) / avgLatency, that directly calculates the relative health factor.

Please feel free to share your thoughts on this.

Reply:
Addressed in the new commit. Please see the comment below.